sdk/src/loom/ModestMaps/Map.ls

```diff
@@ -0,0 +1,713 @@
+ *
+ * @usage <code>
+ *          import loom.modestmaps.Map;
+ *          import loom.modestmaps.geo.Location;
+ *          import loom.modestmaps.mapproviders.BlueMarbleMapProvider;
+ *          ...
+ *          var map:Map = new Map(640, 480, true, new BlueMarbleMapProvider());
+ *          addChild(map);
+ *        </code>
+ *
+ */
+package loom.modestmaps
+{
+    import loom.modestmaps.core.*;
+    import loom.modestmaps.events.*;
+    import loom.modestmaps.geo.*;
+    import loom.modestmaps.mapproviders.IMapProvider;
+    import loom.modestmaps.mapproviders.microsoft.MicrosoftProvider;
+    import loom.modestmaps.overlays.MarkerClip;
+    import loom2d.display.Stage;
+    
+    import loom2d.display.DisplayObject;
+    import loom2d.display.Sprite;
+    import loom2d.events.Event;
+    import loom2d.math.Matrix;
+    import loom2d.math.Point;
+    import loom2d.math.Rectangle;
+    import loom.gameframework.TimeManager;
+    
+    [Event(name="startZooming",      type="loom.modestmaps.events.MapEvent")]
+    [Event(name="stopZooming",       type="loom.modestmaps.events.MapEvent")]
+    [Event(name="zoomedBy",          type="loom.modestmaps.events.MapEvent")]
+    [Event(name="startPanning",      type="loom.modestmaps.events.MapEvent")]
+    [Event(name="stopPanning",       type="loom.modestmaps.events.MapEvent")]
+    [Event(name="panned",            type="loom.modestmaps.events.MapEvent")]
+    [Event(name="resized",           type="loom.modestmaps.events.MapEvent")]
+    [Event(name="mapProviderChanged",type="loom.modestmaps.events.MapEvent")]
+    [Event(name="beginExtentChange", type="loom.modestmaps.events.MapEvent")]
+    [Event(name="extentChanged",     type="loom.modestmaps.events.MapEvent")]
+    [Event(name="beginTileLoading",  type="loom.modestmaps.events.MapEvent")]
+    [Event(name="allTilesLoaded",    type="loom.modestmaps.events.MapEvent")]
+    [Event(name="rendered",          type="loom.modestmaps.events.MapEvent")]
+    [Event(name="markerRollOver",    type="loom.modestmaps.events.MarkerEvent")]
+    [Event(name="markerRollOut",     type="loom.modestmaps.events.MarkerEvent")]
+    [Event(name="markerClick",       type="loom.modestmaps.events.MarkerEvent")]
+    public class Map extends Sprite
+    {
+        protected var mapWidth:Number;
+        protected var mapHeight:Number;
+        protected var __draggable:Boolean = true;
+    
+        /** das grid */
+        public var grid:TileGrid;
+    
+        /** markers are attached here */
+        public var markerClip:MarkerClip;
+        
+        /** Who do we get our Map urls from? How far can we pan? */
+        protected var mapProvider:IMapProvider;
+    
+        /** fraction of width/height to pan panLeft, panRight, panUp, panDown
+         * @default 0.333333333  
+         */
+        public var panFraction:Number = 0.333333333;
+
+        //NOTE_TEC: Added a static Stage because Actionscript Sprites all have their parent stage, whereas Loom does not
```

Comment:
But! What about https://github.com/LoomSDK/LoomSDK/blob/master/sdk/src/loom2d/Display/DisplayObject.ls#L405 ? Not quite the same...